### PR TITLE
Docs: Add Qt Tools to Visual Studio

### PR DESCRIPTION
**Pull request by jwrober**
Docs: Add Qt Tools to Visual Studio. Part of #1400

### CODE REVIEW

File: docs/Contributing/msys2.rst [hunk 1]
```diff
@@ -7,7 +7,8 @@ Setting up MSYS2 for Windows
 ****************************
 
 This document is about building and compiling Freeciv21 with MSYS2 on Windows. MSYS2 is available from
-https://www.msys2.org. The Longturn Community builds the Windows Installer packages using MSYS2.
+https://www.msys2.org. The Longturn Community builds the Windows Installer packages using MSYS2. If you
+are looking for a first class development evironment on Windows, have a look at :doc:`visual-studio`.
```
Comment (daavko) [inline docs/Contributing/msys2.rst:11]:
```suggestion
are looking for a first class development environment on Windows, have a look at :doc:`visual-studio`.
```

File: docs/Contributing/visual-studio.rst [hunk 2]
```diff
@@ -136,6 +136,37 @@ Studio will open a :strong:`CMake Overview Pages` tab.
 Microsoft provides documentation on CMake in Visual Studio --
 https://docs.microsoft.com/en-us/cpp/build/cmake-projects-in-visual-studio?view=msvc-170
 
+Qt Visual Studio Tools
+======================
+
+There is an available extension called :strong:`Qt Visual Studio Tools`. This extension allows you to use the
+Qt Designer and other Qt tools inside of the Visual Studio IDE. This is especially useful if you want to edit
```
Comment (daavko) [inline docs/Contributing/visual-studio.rst:143]:
```suggestion
Qt Designer and other Qt tools within the Visual Studio IDE. This is especially useful if you want to edit
```
I think this sounds better

File: docs/Contributing/visual-studio.rst [hunk 3]
```diff
@@ -136,6 +136,37 @@ Studio will open a :strong:`CMake Overview Pages` tab.
...
+
+Start by `downloading <https://www.qt.io/download-qt-installer>`_ the installer. Double-click the downloaded
+file and login with your Qt account. If you do not have a Qt account, there should be a link in the installer
+window to create one. Click :guilabel:`Next`, agree to the GNU License and state that you are an individual,
+not a business. Click :guilabel:`Next` and pick to either send or not send usage statistics to Qt.
+Click :guilabel:`Next`. Note that installing to :file:`C:\\Qt` is fine. Ensure :guilabel:`Custom Install`
+is selected and then click :guilabel:`Next`. On the right, uncheck :guilabel:`latest support release` and
+check :guilabel:`LTS`, click :guilabel:`Filter`. Expand :guilabel:`Qt` and check the ``5.15.2`` option. Click
+:guilabel:`Next` and agree to the Microsoft license. Click :guilabel:`Next` and note that :strong:`Qt` for
+the Start Menu is fine. Click :guilabel:`Next` and :guilabel:`Install` to begin the process.
```
Comment (daavko) [inline docs/Contributing/visual-studio.rst:154]: Do we really need to describe the whole installation process for Qt here? I'd just mention the parts of Qt that are required for this to work, not which exact buttons to press

File: docs/Contributing/visual-studio.rst [hunk 4]
```diff
@@ -136,6 +136,37 @@ Studio will open a :strong:`CMake Overview Pages` tab.
...
+.. note:: You can reduce the size of the Qt Tools install by expanding the ``5.15.2`` option and unchecking
+  ``WebAssembly``, ``MSVC 2015 64-bit``, ``MSVC 2019 32-bit``, ``MinGW 8.1.0 32-bit``, ``MinGW 8.1.0 64-bit``,
+  ``UWP*``, and ``Android``. Unless you intend to develop for those platforms, you do not need to download and
+  install those components.
+
+Now to install the extension. In Visual Studio, click on
+:menuselection:`Extensions --> Manage Extensions --> Online`. In the upper right of the window, you will see
+a search box. Search for :strong:`Qt Visual Studio Tools`. Select it in the center and click the
+:guilabel:`Install` button. When it is finished downloading, close Visual Studio. A new window will
+come up that installs the extension. When finished re-open Visual Studio.
```
Comment (daavko) [inline docs/Contributing/visual-studio.rst:165]: Again, this seems a little verbose. I might be biased, but I'd expect a developer to know how to install extensions in their IDE.

**Review by daavko — COMMENTED** [review]
Just a couple notes

File: docs/Contributing/visual-studio.rst (hunk 3 above)
Comment (jwrober) [reply docs/Contributing/visual-studio.rst:154]: I'm pretty specific on exactly what steps to take to get base VS installed and configured, so I am following the same here.

File: docs/Contributing/visual-studio.rst (hunk 4 above)
Comment (jwrober) [reply docs/Contributing/visual-studio.rst:165]: I shrunk this one down though.

**Pull request merged**